src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java

```diff
@@ -457,10 +462,31 @@ public ByteBuffer getResourceBuffer(ImageLocation loc) {
         return null;
     }
 
-    public InputStream getResourceStream(ImageLocation loc) {
-        Objects.requireNonNull(loc);
-        byte[] bytes = getResource(loc);
+    /**
+     * Specialized {@link IOException} thrown during construction to provide a
+     * semantic reason for failure and allow better user-facing error messages.
+     */
+    public final static class ImageError extends IOException {
```

Comment:
Exception names ending in "Error" are usually fatal to the VM, this seems more like a normal exception.
Generally, we don't add single purpose exceptions, they are little used and just contribute to bloat.